### ISSUE TITLE
[NV-1178] MSTeams integration throwing error

### ISSUE THREAD

**Issue by scopsy** [novuhq/novu#2026]
### What?

As reported by a community member on [Discord](https://discord.com/channels/895029566685462578/1041671116063780864/1041671116063780864) sending an MS team notification throws an error:

```
{
  "stack": "SyntaxError: Unexpected token z in JSON at position 0\n    at JSON.parse (<anonymous>)\n    at MsTeamsProvider.<anonymous> (/usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:57:86)\n    at step (/usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:33:23)\n    at Object.next (/usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:14:53)\n    at /usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:8:71\n    at new Promise (<anonymous>)\n    at __awaiter (/usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:4:12)\n    at MsTeamsProvider.sendMessage (/usr/src/app/providers/ms-teams/build/main/lib/msTeams.provider.js:53:16)\n    at MSTeamsHandler.<anonymous> (/usr/src/app/apps/api/dist/main.js:15864:40)\n    at Generator.next (<anonymous>)\n    at /usr/src/app/apps/api/dist/main.js:15846:71\n    at new Promise (<anonymous>)\n    at __awaiter (/usr/src/app/apps/api/dist/main.js:15842:12)\n    at MSTeamsHandler.send (/usr/src/app/apps/api/dist/main.js:15860:16)\n    at SendMessageChat.<anonymous> (/usr/src/app/apps/api/dist/main.js:15750:50)\n    at Generator.next (<anonymous>)",
  "message": "Unexpected token z in JSON at position 0"
}
```

 The credentials provided:

```
{
      "providerId": "msteams",
      "credentials": {
          "webhookUrl":"url"
      }
    }
```

> From [SyncLinear.com](https://synclinear.com) | [NV-1178](https://linear.app/novu/issue/NV-1178/msteams-integration-throwing-error)

**Comment by dcamarasa** [comment]
The issue was not on the provider but on the format of the message content that was being sent. MS Teams enforces the messages to be in JSON format.

**Comment by ducdung8491** [comment]
I fixed it #2029

**Comment by dcamarasa** [comment]
Sorry @ducdung8491, but in the fix you are supposing that messages should be put in a "text" field in the JSON message to MS teams, but the json can contain other several fields like 'summary', '@type', 'contentType', etc. That's why the content you put in the notification template should already contain the JSON you want to be sent.

Take a look at connector card examples: https://learn.microsoft.com/en-us/microsoftteams/platform/task-modules-and-cards/cards/cards-reference

**Comment by ducdung8491** [comment]
@dcamarasa what you say is actionable messages. MS webhook also support [simple message](https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/how-to/connectors-using?tabs=cURL#send-messages-using-curl-and-powershell)

[Chat message](https://github.com/novuhq/novu/blob/271488a5ea716f00167d397c01694e8287d5c289/packages/stateless/src/lib/provider/provider.interface.ts#L57-L60) send to provider only content.

**Comment by dcamarasa** [comment]
But with this solution then the provider would only send simple messages. Actionable messages wouldn't be supported.

**Comment by ducdung8491** [comment]
I think you should create new a new issue if you would like to support Actionable messages

**Comment by ducdung8491** [comment]
ok, I have got it

**Comment by ducdung8491** [comment]
Hi @dcamarasa. I have an update.

**Comment by scopsy** [comment]
Closed in favor of #2029 the merged PR.
